### PR TITLE
README.md running from source is a killer feature

### PR DESCRIPTION
**Pull request by techtonik**
The rationale is based on assumption that README.md is written for GitHub users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -149,17 +157,6 @@ greater than 2.6.
 
 * Pep8 (style analysis).
 
-## Running from source
-
-It is possible to run Spyder directly (i.e. without installation)
-from the unpacked zip folder (see *Installing from source*) using
-Spyder's bootstrap script like this:
```
Comment (Nodd) [inline README.md]: unpacked zip folder or git repository!
Also, having a reference to "Installing from source" right after saying that you don't need to install it is a bit disturbing, but maybe it's only me.

File: README.md [hunk 2]
```diff
@@ -44,6 +44,14 @@ Spyder may also be used as a PyQt4/PySide extension library (module
 `spyderlib`). For example, the Python interactive shell widget used in
 Spyder may be embedded in your own PyQt4/PySide application.
 
+## Running from source
```
Comment (ccordoba12) [inline README.md]: Why put this section here? I think the place where it was placed was better than this one.

We want to tell people how to install Spyder first, not how to run it from source.

File: README.md (hunk 1 above)
Comment (Nodd) [inline README.md]: Uh, why did I comment on the removed part ? :confused:

File: README.md [hunk 3]
```diff
@@ -44,6 +44,14 @@ Spyder may also be used as a PyQt4/PySide extension library (module
...
+
+It is possible to run Spyder from the source code checkout, given
+that you have PyQt or PySide installed. Just run:
```
Comment (Nodd) [inline README.md]: Why the reference to PyQt and PySide here ? Spyder also needs numpy, shinx, and others, right ?

Comment (ccordoba12) [inline README.md]: PyQt is the minimum needed. All other modules are optional.

File: README.md (hunk 1 above)
Comment (techtonik) [inline README.md]: Did you know about the feature before?

File: README.md (hunk 2 above)
Comment (techtonik) [inline README.md]: Who are we? As I already said, the README.md is for GitHub users, who prefer to work with sources if possible. For everybody else there are other other channels.

Cloning and running is that **fastest** way a person from GitHub can run Spyder. I don't see why we want to put users through the installation rituals.

**Comment by techtonik** [comment]
Added more edits to omit obvious parts and get new/important information to the readers faster.

File: README.md (hunk 2 above)
Comment (ccordoba12) [inline README.md]: I don't know, maybe using the Readme to help Github users is the best thing to do in 2015 :-)

@Nodd, what do you think?

Comment (ccordoba12) [inline README.md]: If others can comment on this one, that'd be great!

@goanpeca, @blink1073, @SylvainCorlay, what do you think?

File: README.md [hunk 4]
```diff
@@ -76,26 +87,23 @@ The easiest way to install Spyder is:
     - Using the [Anaconda Distribution](http://continuum.io/downloads).
     - Installing from source (see below).
 
-### Installing from source
+* Cross-platoform way from source
```
Comment (blink1073) [inline README.md]: platform*

File: README.md (hunk 2 above)
Comment (blink1073) [inline README.md]: @ccordoba12, I agree with taking a developer-first stance on a Github readme, as long as we also mention `pip install` somewhere.

File: README.md (hunk 1 above)
Comment (Nodd) [inline README.md]: Yes.

File: README.md [hunk 5]
```diff
@@ -44,6 +44,17 @@ Spyder may also be used as a PyQt4/PySide extension library (module
...
+
+The fastest way to run Spyder is to get the source code, install PyQt
+or PySide, and run:
+
+    python bootstrap.py
+
+You may want to do this for fixing bugs in Spyder, adding new
+features, learning how Spyder works or just getting a taste of it.
+Make sure you use a stable branch though, because `master` may not
+always be clean.
```
Comment (Nodd) [inline README.md]: master is not always clean, but I can't remember a time where spyder didn't work at all. The precision here seems too much.

File: README.md [hunk 6]
```diff
@@ -76,26 +87,23 @@ The easiest way to install Spyder is:
...
 
     python setup.py install
 
-Note that `distutils` does *not* uninstall previous versions of Python 
-packages: it simply copies files on top of an existing installation. 
-When using this command, it is thus highly recommended to uninstall 
-manually any previous version of Spyder by removing the associated 
-directories ('spyderlib' and 'spyderplugins') from your site-packages 
-directory).
+Note that `setup.py` script doens't know how uninstall previous versions
```
Comment (Nodd) [inline README.md]: how **to** uninstall

**Comment by Nodd** [comment]
BTW python can run modules from zip files. Is in the case for spyder ? I think it requires a __main__.py or something...

File: README.md (hunk 2 above)
Comment (techtonik) [inline README.md]: If there is user level docs, it is possible to add a redirect to them at the top of README.md The top also needs a link to wiki KB.

File: README.md (hunk 4 above)
Comment (techtonik) [inline README.md]: Fixed.

File: README.md (hunk 5 above)
Comment (techtonik) [inline README.md]: Removed stuff about master. Maybe it is just my system, but current master is extremely slow.

File: README.md (hunk 6 above)
Comment (techtonik) [inline README.md]: Fixed.

**Comment by techtonik** [comment]
I can try to wrap an executable Spyder .zip, but I am not sure that pip will be able to install it - https://github.com/pypa/pip/issues/2406

File: README.md (hunk 5 above)
Comment (Nodd) [inline README.md]: You should report it then !

**Comment by Nodd** [comment]
That would require to create a separate executable zip file. Not worth the hassle, it was just a random suggestion.

**Comment by ccordoba12** [comment]
I think this is ready. Thanks for the update to our README @techtonik. It was really needed :-)

Merging then.

**Pull request merged**